feat: install Grafana, Prometheus, Harvest

cgrinds opened the pull request:
- Added `docker generate full` command to stand-up Prometheus, Grafana, and a container per poller
- Instead of duplicating the dashboards, I moved them and updated references
    - Verfied that `bin/grafana import` works with new directory structure
- Removed allPollerInOneContainer
- Updated dashboards to include datasource template for auto-install (provisioning)

rahulguptajss commented on docker/README.md:
```diff
@@ -0,0 +1,106 @@
+
+## Config and Using
+
+The `prom-stack.yml` compose file creates a `frontend` and `backend` network. Prometheus and Grafana publish their admin ports on the front-end network and are routable to the local machine. By default, the Harvest pollers are part of the backend network and do not expose their Prometheus web end-points. If you want their end-points exposed, pass the `--ports` flag to `generate` like so:
```
should be `--port` instead of `--ports`

cgrinds replied:
nice catch!

rahulguptajss commented on docker/README.md:
````diff
@@ -0,0 +1,106 @@
+#### Stop all containers
+
+```
+docker compose -f docker/prom-stack.yml -f harvest-compose.yml down
````
docker-compose

rahulguptajss commented on docker/onePollerPerContainer/docker-compose.tmpl:
```diff
@@ -1,23 +1,25 @@
-version: "3.5"
```
If i recall kompose (k8 artifacts) were not generated without 3+ version. pls do check once.

cgrinds replied:
Yeah it's a bit of a [mess](https://docs.docker.com/compose/compose-file/compose-versioning/#versioning). Sticking with `3.5` seems reasonable.  The reason I wanted to remove it doesn't exist anymore so I'll add it back

cgrinds commented on docker/README.md:
````diff
@@ -0,0 +1,106 @@
+Bring everything up :tada:
+   
+```
+docker compose -f docker/prom-stack.yml -f harvest-compose.yml up -d --remove-orphans
````
Ah this is a platform difference - compose was rolled into docker for Mac and Windows, but not Linux. The plan is to replace docker-compose everywhere but not yet. Good catch.

cgrinds commented on cmd/tools/generate/generate.go:
```diff
@@ -61,40 +77,83 @@ func doDockerCompose(cmd *cobra.Command, _ []string) {
+			Port:          port,
+			LogLevel:      opts.loglevel,
+			Image:         opts.image,
+			ContainerName: "poller_" + v,
```
it's somewhat handy, but it doesn't work in dev since the version is 2.0.2

cgrinds commented:
- re-added version to container name
- improved volume support so it works on all platforms (sudo docker on Linux did not work with PWD)
- replaced beta `docker compose` with `docker-compose`
- moved `prom-stack.yml` from docker directory to root so host path resolution work correctly on Linux
- rebased to incorporate Hardik's dashboard fixes

rahulguptajss commented on docker/onePollerPerContainer/docker-compose.tmpl:
```diff
@@ -1,23 +1,27 @@
       - harvest:/opt/harvest/
       - type: bind
-        source: {{ .ConfigFile }}
+        source: ./{{ .ConfigFile }}
```
Why relative path? We want absolute path here.

cgrinds replied:
[Docker](https://stackoverflow.com/questions/42248198/how-to-mount-a-single-file-in-a-volume/59554543#59554543) turns this into an absolute path - verified on linux and macos
